File: src/main/java/model/equipment/weapon/TwoHandedHammer.java
package model.equipment.weapon;

import model.support.allEnums.DmgType;
import model.support.allEnums.WeaponType;

public class TwoHandedHammer extends Weapon {
    public TwoHandedHammer(double weight, int quality) {
        super(WeaponType.TWO_HANDED_HAMMER, weight, quality);
        setDamage();
    }

    private void setDamage() {
        damage.put(DmgType.BLUNT, quality * 0.65);
        damage.put(DmgType.PIERCING, quality * 0.35);
    }
}

File: src/main/java/model/equipment/weapon/TwoHandedHammer.kt
package model.equipment.weapon

import model.support.enums.Skill
import model.support.enums.Skill.TWO_HANDED_HAMMER_SKILL
import model.support.enums.Skill.TWO_HANDED_WEAPON_SKILL
import model.support.enums.Specification.*
import model.support.enums.WeaponType.TWO_HANDED_HAMMER

class TwoHandedHammer(weight: Double, quality: Int) : TwoHandedWeapon(TWO_HANDED_HAMMER, TWO_HANDED_HAMMER_SKILL, TWO_HANDED_WEAPON_SKILL, weight, quality, false) {
    init {
        damage = mapOf(
                BLUNT to quality * 0.65,
                CUT to 0.0,
                SLASH to 0.0,
                PIERCING to quality * 0.35,
                CRUSH to 0.0)

    }
}
